src/transformers/training_args.py

```diff
@@ -1934,29 +1938,31 @@ def __post_init__(self):
                 elif fsdp_option == FSDPOption.AUTO_WRAP:
                     os.environ[f"{prefix}AUTO_WRAP_POLICY"] = FSDP_AUTO_WRAP_POLICY[0]
-                    if self.fsdp_config["min_num_params"] > 0:
-                        os.environ[f"{prefix}MIN_NUM_PARAMS"] = str(self.fsdp_config["min_num_params"])
+                    if self.fsdp_config.min_num_params > 0:
+                        os.environ[f"{prefix}MIN_NUM_PARAMS"] = str(self.fsdp_config.min_num_params)
                         os.environ[f"{prefix}AUTO_WRAP_POLICY"] = FSDP_AUTO_WRAP_POLICY[1]
-                    elif self.fsdp_config.get("transformer_layer_cls_to_wrap", None) is not None:
+                    elif self.fsdp_config.transformer_layer_cls_to_wrap is not None:
                         os.environ[f"{prefix}TRANSFORMER_CLS_TO_WRAP"] = ",".join(
-                            self.fsdp_config["transformer_layer_cls_to_wrap"]
+                            self.fsdp_config.transformer_layer_cls_to_wrap
                         )
-            prefetch_policy = self.fsdp_config.get("backward_prefetch", "NO_PREFETCH")
+            prefetch_policy = self.fsdp_config.backward_prefetch
             os.environ[f"{prefix}BACKWARD_PREFETCH"] = prefetch_policy.upper()
-            os.environ[f"{prefix}FORWARD_PREFETCH"] = str(self.fsdp_config.get("forward_prefetch", "false")).lower()
+            os.environ[f"{prefix}FORWARD_PREFETCH"] = str(self.fsdp_config.forward_prefetch).lower()
 
-            sync_module_states = str(self.fsdp_config.get("sync_module_states", "true")).lower()
-            cpu_ram_efficient_loading = str(self.fsdp_config.get("cpu_ram_efficient_loading", "false")).lower()
+            sync_module_states = str(self.fsdp_config.sync_module_states).lower()
+            cpu_ram_efficient_loading = str(self.fsdp_config.cpu_ram_efficient_loading).lower()
 
             if sync_module_states == "false" and cpu_ram_efficient_loading == "true":
                 # In this case, all the processes except the main process would have random weights leading
                 # to unexpected behaviour during training, thus throwing error here to prevent it.
-                raise ValueError('`sync_module_states` must be `"True"` if `cpu_ram_efficient_loading` is `"True"`')
+                raise ValueError("`sync_module_states` must be `True` if `cpu_ram_efficient_loading` is `True`")
 
             os.environ[f"{prefix}SYNC_MODULE_STATES"] = sync_module_states
             os.environ[f"{prefix}CPU_RAM_EFFICIENT_LOADING"] = cpu_ram_efficient_loading
 
-            os.environ[f"{prefix}USE_ORIG_PARAMS"] = str(self.fsdp_config.get("use_orig_params", "true")).lower()
+            os.environ[f"{prefix}USE_ORIG_PARAMS"] = str(self.fsdp_config.use_orig_params).lower()
+            os.environ[f"{prefix}USE_ORIG_PARAMS"] = str(self.fsdp_config.use_orig_params).lower()
+            os.environ[f"{prefix}STATE_DICT_TYPE"] = self.fsdp_config.state_dict_type.upper()
```

Comment:
### Code change comments


Support state_dict_type by propagating that info to FSDP plugin.